internal/storage/fake/bucket.go

```diff
@@ -973,6 +1004,25 @@ func (b *bucket) CreateFolder(ctx context.Context, folderName string) (*gcs.Fold
 		sort.Sort(b.folders)
 	}
 
+	// In the hierarchical bucket, control client API creates folders  as well as
+	// prefixes for backward compatibility. Therefore, a prefix object
+	// entry needs to be created here as well.
+
+	// Find any existing record for this name.
+	existingObjectPrefixIndex := b.objects.find(folderName)
+
+	// Create a prefix object record from the given attributes.
+	var prefixObject fakeObject
+	prefixObject.metadata = gcs.Object{Name: folderName}
+
+	// Replace an entry in or add an entry to our list of objects.
+	if existingObjectPrefixIndex < len(b.objects) {
+		b.objects[existingIndex] = prefixObject
```

Comment:
are you supposed to use existingObjectPrefixIndex here?

Reply:
Done thanks.